lib/trie.js

```diff
@@ -0,0 +1,66 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function(path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!(parts[i] in node)) {
+        node[parts[i]] = new Trie();
+      }
+      if (i + 1 === parts.length) {
+        addVerbHandler(node[parts[i]], verb, handler);
+      }
+      node = node[parts[i]];
+    }
+  } else { // if we would ever encounter this situation?
+    addVerbHandler(node, verb, handler);
+  }
+  return this;
+};
+
+//Look for matching path and return handler method for a match
+Trie.prototype.find = function(path) {
```

Comment:
@bajtos: In our implementation, `find` is only accepting the `path` as it's being called by `RestRouter.handle()`. `find` returns a `methods` object which has all `{verb:handlers}` registered for that path. 
I'm using `RestRouter.handle()` to do the matching work for `verb`. This way, we can keep our matching criteria for `verb+path` (such as, we may need to handle multiple handler for same verb+path) separated from trie implementation.

For the sake of consistency, I've changed the `trie.add(path, verb, handler)` to `trie.add(route, handler)`
and have kept `trie.find(path)` as it is.